README.md

```diff
@@ -905,6 +905,60 @@ BENCHMARK(BM_test)->Range(8, 8<<10)->UseRealTime();
 
 Without `UseRealTime`, CPU time is used by default.
 
+Google/benchmark uses `std::thread` as multithreading environment per default.
+If you want to use another multithreading environment (e.g. OpenMP), you can 
+turn off the automatic threading using the `ManualThreading` function.
+```c++
+static void BM_MultiThreaded(benchmark::State& state) {
+  // Setup code here.
+  for (auto _ : state) {
+#pragma omp parallel num_threads(state.threads)
+    // Run the multithreaded test. 
+  }
+  // Teardown code here.
+}
+
+BENCHMARK(BM_MultiThreaded)->ManualThreading()->Threads(1)->Threads(2)->Threads(4);
+```
+The above example creates a parallel region in each iteration. 
+This includes the setup and teardown of the parallel region in the time measurment and it 
+adds an implicit barrier at the end of each iteration.   
+You can avoid these effects, if you run the whole loop in parallel. 
+Then you must not use the `state` object directly, but create a `ThreadState` object in each thread.
+```c++
+static void BM_MultiThreaded(benchmark::State& state) {
+  // Setup code (shared objects) here.
+#pragma omp parallel num_threads(state.threads)
+  {
+    // Thread-local setup code here.
+    for (auto _ : benchmark::ThreadState(state)) {
+      // Run the multithreaded test. 
+    }
+  }
+  // Teardown code here.
+}
+
+BENCHMARK(BM_MultiThreaded)->ManualThreading()->Threads(1)->Threads(2)->Threads(4);
+```
+If you use the `ThreadState` object and explicitly specify the number of threads, then you must 
+use `ManualThreading` and the number of created `ThreadState` objects must match the number of specified threads. 
```

Comment:
Is this postcondition actually required? Can it be lifted?

Reply:
Do you mean the postcondition, that number of ThreadStates == number of threads? I don't see a way to do this. 
Note, that this postcondition is only requried, if you explicitly specify the number of threads via `Threads()` and if you create at least one `ThreadState` object during the benchmark run. It would be odd, if in this case the numbers doesn't match.

Reply:
> Do you mean the postcondition, that number of ThreadStates == number of threads?

Yes.

> I don't see a way to do this.
> Note, that this postcondition is only requried, if you explicitly specify the number of threads via `Threads()` and if you create at least one `ThreadState` object during the benchmark run. It would be odd, if in this case the numbers doesn't match.

In particular, as discussed (?) in previous PR, i would like to call `->Threads()`,
but disable `benchmark`-based threading. I.e. i'd like a way to use threads
as yet another parameter, without it actually creating all those threads,
and without dividing the measurements by the "thread count".